pkg/processor/k8sprocessor/pod_association.go

```diff
@@ -87,20 +92,22 @@ func extractPodID(ctx context.Context, attrs pcommon.Map, associations []kube.As
 			// Build hostname from pod k8s.pod.name and k8s.namespace.name attributes
 			pod, ok := attrs.Get(conventions.AttributeK8SPodName)
 			if !ok {
-				return "", ""
+				return "", "", errors.New("pod name not found in attributes")
 			}
 
 			namespace, ok := attrs.Get(conventions.AttributeK8SNamespaceName)
 			if !ok {
-				return "", ""
+				return "", "", errors.New("namespace name not found in attributes")
 			}
 
 			if pod.StringVal() == "" || namespace.StringVal() != "" {
-				return asso.Name, kube.PodIdentifier(fmt.Sprintf("%s.%s", pod.StringVal(), namespace.StringVal()))
+				podIdentifierKey = asso.Name
+				podIdentifierValue = kube.PodIdentifier(fmt.Sprintf("%s.%s", pod.StringVal(), namespace.StringVal()))
+				return
 			}
 		}
 	}
-	return "", kube.PodIdentifier("")
+	return "", "", errors.New("could not assign pod id basing on associations")
```

Comment:
Is there a difference between an empty string and `kube.PodIdentifier("")` since `kube.PodIdentifier` is defined as string?

Reply:
There isn't in practice, but I'd personally prefer using the full type name `kube.PodIdentifier("")`.

This is a bit subtle, because you can do `return "", ""`, but if you did:
```go
var retval string = ""
return "", retval
```
you'd get an error due to type mismatch. Go does a bit of implicit type conversion in specific circumstances, and this is one of them.